File: PYTHON/Concesionario/transaccionVentaTaller.py
class TransaccionVentaTaller(Transaccion):
    transaccionesvental = []

    def __init__(self, tipo, ingreso, cliente, articulo, vendedor, transfer):
        super().__init__(tipo, ingreso, cliente, transfer)
        self.articulo = articulo
        self.vendedor = vendedor
        TransaccionVentaTaller.transaccionesvental.append(self)

    def info(self):
        txt = f"Transacción #{self.transfer:08d}: venta realizada por {self.vendedor.getNombre()} para el cliente {self.cliente.getNombre()} por un total de ${self.articulo.getPrecio()} por el artículo {self.articulo.getMarca()}"
        return txt

    @staticmethod
    def getClientePorCedula(cedula):
        finder = None
        cli = None

        for trans in TransaccionVentaTaller.transaccionesvental:
            if trans.getClienteCed() == cedula:
                finder = trans
                cli = finder.getCliente()
                break

        return cli

    @staticmethod
    def getTransaccionporCedula(cedula):
        finder = None

        for trans in TransaccionVentaTaller.transaccionesvental:
            if trans.getClienteCed() == cedula:
                finder = trans.getArticulo()
                break

        return finder

    def setArticulo(self, articulo):
        self.articulo = articulo

    def getArticulo(self):
        return self.articulo

    def setVendedor(self, vendedor):
        self.vendedor = vendedor

    def getVendedor(self):
        return self.vendedor

    @staticmethod
    def agregarTransaccion(transaccion):
        TransaccionVentaTaller.transaccionesvental.append(transaccion)

    @staticmethod
    def eliminarTransaccion(transaccion):
        TransaccionVentaTaller.transaccionesvental.remove(transaccion)

    @staticmethod
    def getTransaccionesven():
        return TransaccionVentaTaller.transaccionesvental

File: Concesionario/src/gestorAplicacion/activos/TransaccionVentaTaller.java
package gestorAplicacion.activos;

import java.io.Serializable;
import java.util.ArrayList;

import gestorAplicacion.personal.*;

public class TransaccionVentaTaller extends Transaccion implements Serializable{
	private static final long serialVersionUID = 1L;
	private Articulo articulo;
	private Vendedor vendedor;
	static ArrayList<TransaccionVentaTaller> transaccionesvental = new ArrayList<TransaccionVentaTaller>();
	
	public TransaccionVentaTaller(String tipo, long  ingreso, Cliente cliente, Articulo articulo, Vendedor vendedor, int transfer) {
		super(tipo, ingreso, cliente, transfer);
		this.articulo=articulo;
		this.vendedor=vendedor;
		TransaccionVentaTaller.transaccionesvental.add(this);
	}
	@Override
	public String info() {
		 String txt = String.format("Transacción #%08d: venta realizada por %s para el cliente %s por un total de $%d por el articulo %s",
		            transfer, vendedor.getNombre(), cliente.getNombre(), articulo.getPrecio(), articulo.getMarca());
			
		
		return txt;
	}
	public static Cliente getClientePorCedula(long cedula){
		Transaccion finder = null;
		Cliente cli=null;
		
		for(Transaccion trans: transaccionesvental) {
			if(trans.getClienteCed()== cedula) {
				finder = trans;
				cli=finder.getCliente();
				break;
			}
		}
		return cli;
	}
	public static Articulo getTransaccionporCedula(long cedula) {
		Articulo finder = null;
		for (TransaccionVentaTaller trans: transaccionesvental) {
			if(trans.getClienteCed()== cedula) {
				finder = trans.getArticulo();
				break;
			}
		}
		return finder;
	}

	public void setArticulo(Articulo articulo) {
	    this.articulo = articulo;
	}

	public Articulo getArticulo() {
	    return articulo;
	}
	public void setVendedor(Vendedor vendedor) {
	    this.vendedor = vendedor;
	}

	public Vendedor getVendedor() {
	    return vendedor;
	}
	public static void agregarTransaccion(TransaccionVentaTaller transaccion) {
	    transaccionesvental.add(transaccion);
	}

	public static void eliminarTransaccion(TransaccionVentaTaller transaccion) {
	    transaccionesvental.remove(transaccion);
	}

	public static ArrayList<TransaccionVentaTaller> getTransaccionesven() {
	    return transaccionesvental;
	}
	
}

